eppo_core/src/event_ingestion/event_dispatcher.rs

```diff
@@ -0,0 +1,219 @@
+use crate::event_ingestion::batched_message::BatchedMessage;
+use crate::event_ingestion::delivery::DeliveryStatus;
+use crate::event_ingestion::event::Event;
+use crate::event_ingestion::event_delivery::EventDelivery;
+use crate::event_ingestion::queued_event::{QueuedEvent, QueuedEventStatus};
+use crate::event_ingestion::vec_event_queue::{EventQueue, QueueError};
+use crate::event_ingestion::{auto_flusher, batcher, delivery};
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::{Receiver, Sender, UnboundedSender};
+use tokio::time::Duration;
+use url::Url;
+
+#[derive(Debug)]
+pub enum EventDispatcherCommand {
+    Event,
+}
+
+// batch size of one means each event will be delivered individually, thus effectively disabling batching.
+const MIN_BATCH_SIZE: usize = 1;
+const MAX_BATCH_SIZE: usize = 10_000;
+
+#[derive(Debug, Clone)]
+pub struct EventDispatcherConfig {
+    pub sdk_key: String,
+    pub ingestion_url: String,
+    pub delivery_interval: Duration,
+    pub retry_interval: Duration,
+    pub max_retry_delay: Duration,
+    pub batch_size: usize,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum DispatcherError {
+    #[error("Queue error: {0}")]
+    QueueError(QueueError),
+    #[error("Receiver should not be closed before all senders are closed")]
+    EventDeliveryError,
+}
+
+/// EventDispatcher is responsible for batching events and delivering them to the ingestion service
+/// via [`EventDelivery`].
+pub struct EventDispatcher<T> {
+    config: EventDispatcherConfig,
+    event_queue: T,
+}
```

Comment:
Event dispatcher should be a thin wrapper around `Sender<BatchedMessage<Event>>` now